File: 20190511_diverta2019PC/B-RGB_Boxes.cpp
#include <iostream>
using namespace std;

int main() {
    int R, G, B, N; cin >> R >> G >> B >> N;
    int n_r = N / R, n_g = N / G;
    int b, r_b, count = 0;
    for (int r = 0; r <= n_r; r++) {
        for (int g = 0; g <= n_g; g++) {
            b = (N - R * r - G * g) / B;
            r_b = (N - R * r - G * g) % B;
            if (b >= 0 && r_b == 0) count++;
        }
    }
    cout << count << endl;
    return 0;
}

File: 20190511_diverta2019PC/B-RGB_Boxes.py
def main():
    R, G, B, N = map(int, input().split())
    n_r = int(N / R)
    n_g = int(N / G)
    count = 0
    for r in range(n_r+1):
        for g in range(n_g+1):
            b = (N - R * r - G * g) // B
            r_b = (N - R * r - G * g) % B
            if b >= 0 and r_b == 0:
                count += 1
    print(count)


if __name__ == '__main__':
    main()
